tests/test_cli.py

```diff
@@ -0,0 +1,174 @@
+    :return: None
+    """
+    subprocess.run(["./grinder.py"])
+    assert "Usage: ./grinder.py -h for help" in str(capfd.readouterr().out)
+    subprocess.run(cli_args)
+    out, err = capfd.readouterr()
+    assert (
+        "Oops! File with queries was not found. Create it or set name properly."
+        in str(out)
+    )
+    assert "No such file or directory: 'queries.json'" in str(out)
+
+
+def test_cli_invalid_base_args(capfd) -> None:
+    """
+    Check behavior in case of invalid value following flag that doesn't require value
+    :return: None
+    """
+    base_one = ["./grinder.py", "-r"]
+    flags = ["-r", "-u", "-cu", "-cp", "-ni"]
+    invalid_value = "wrong_one"
+    for f in flags:
+        base_one.extend([f, invalid_value])
+        subprocess.run(base_one)
+        out, err = capfd.readouterr()
+        assert "unrecognized arguments:" in str(err)
+        base_one.pop()
+        subprocess.run(base_one)
+        out, err = capfd.readouterr()
+        assert (
+            "Oops! File with queries was not found. Create it or set name properly."
+            in str(out)
+        )
+        base_one.pop()
+
+
+def test_cli_invalid_args(cli_args: list, capfd) -> None:
+    """
+    Check behavior in case of invalid args with different flags
+    :param cli_args: base arguments
+    :return: None
+    """
+    base_args = cli_args
+    invalid_value = "wrong_one"
+    invalid_args = base_args
+
+    invalid_args.append(invalid_value)
+    subprocess.run(invalid_args)
+    out, err = capfd.readouterr()
+    assert "unrecognized arguments:" in str(err)
+
+    invalid_args = base_args
+    invalid_args.extend(["-nm", "-vs", "-sc", "-vs", "-ts"])
+    flags = ["-cm", "-sm", "-nw", "-vw", "-ht", "-tp", "-ml", "-q", "-v", "-tsp", "-vk"]
+
+    expected_invalid_errors = [
+        "argument -cm/--censys-max: invalid int value",
+        "argument -sm/--shodan-max: invalid int value",
+        "argument -nw/--nmap-workers: invalid int value",
+        "argument -vw/--vulners-workers: invalid int value",
+        "argument -ht/--host-timeout: invalid int value",
+        "argument -tp/--top-ports: invalid int value",
+        "argument -ml/--max-limit: invalid int value",
+        "unrecognized arguments:",
+        "unrecognized arguments:",
+        "unrecognized arguments:",
+        "unrecognized arguments:",
+    ]
+
+    for i in range(len(flags)):
+        invalid_args.extend([flags[i], invalid_value])
+        subprocess.run(invalid_args)
+        err1 = capfd.readouterr().err
+        assert expected_invalid_errors[i] in str(err1)
+
+        invalid_args.pop()
+        subprocess.run(invalid_args)
+        out, err = capfd.readouterr()
+        if str(err) != str(err1):
+            assert "expected one argument" in str(err)
+        invalid_args.pop()
+
+
+def test_cli_vendor_confidence(cli_args: list, query_random: str, capfd) -> None:
+    """
+    Check behavior in case of different vendor confidence
```

Comment:
For this purpose, if you want to check behavior more accurate, you can use `-d / --debug` flag that will show you additional information in output.

Example output:
```
  .,-:::::/ :::::::..   ::::::.    :::.:::::::-.  .,:::::: :::::::..
,;;-'````'  ;;;;``;;;;  ;;;`;;;;,  `;;; ;;,   `';,;;;;'''' ;;;;``;;;;
[[[   [[[[[[/[[[,/[[['  [[[  [[[[[. '[[ `[[     [[ [[cccc   [[[,/[[['
"$$c.    "$$ $$$$$$c    $$$  $$$ "Y$c$$  $$,    $$ $$""""   $$$$$$c
 `Y8bo,,,o88o888b "88bo,888  888    Y88  888_,o8P' 888oo,__ 888b "88bo,
   `'YMUP"YMMMMMM   "W" MMM  MMM     YM  MMMMP"`   """"YUMMMMMMM   "W"

Shodan API key: YOUR_DEFAULT_API_KEY
Censys API ID: YOUR_CENSYS_API_ID
Censys API SECRET: YOUR_CENSYS_API_SECRET
Vulners API key: YOUR_VULNERS_API_KEY_HERE
Query confidence level: all queries, any confidence
Vendor confidence level: all vendors, any confidence
Vendors to scan: all vendors
Shodan max results quantity: None
Censys max results quantity: None
File with queries: queries/test.json
```